src/frontend/e2e/pages/filter.ts

```diff
@@ -1,103 +1,96 @@
-import { Page } from "@playwright/test";
-export abstract class FilterSample {
-  public static async applyFilter(
-    page: Page,
-    filterData: Partial<FilterData>
-  ): Promise<void> {
-    // wait for sample page is ready to be handled
-    await page.waitForFunction(() => {
-      const samples = document.querySelector(
-        "a[href$='/data/samples'] > div > div:nth-child(2)"
-      )?.textContent;
-      return parseInt(samples!) > 0;
-    });
-    if (
-      filterData.uploadDateFrom !== undefined ||
-      filterData.uploadDateTo !== undefined
-    ) {
-      await page.locator("button[label='Upload Date']").click();
-      if (filterData.uploadDateFrom !== undefined) {
-        await page
-          .locator("input[name='uploadDateStart']")
-          .fill(filterData.uploadDateFrom);
-      }
-      if (filterData.uploadDateTo !== undefined) {
-        await page
-          .locator("input[name='uploadDateEnd']")
-          .fill(filterData.uploadDateTo);
-      }
-      await page
-        .locator(
-          "//div[not(contains(@style,'visibility: hidden')) and contains(@class,'MuiPaper-root')]/descendant::button[text()='Apply']"
-        )
-        .click();
-    }
-    // select upload date period
-    if (filterData.uploadDatePeriod !== undefined) {
-      await page.locator("button[label='Upload Date']").click();
-      await page
-        .locator("div:not([style*='hidden'])[class*='MuiPaper-elevation'] li")
-        .filter({ hasText: filterData.uploadDatePeriod })
-        .click();
+import { BasePage } from "../pages/basePage";
+
+// until we have reliable data-test-id, we will use
```

Comment:
are reliable data-test-ids coming in future PRs?

Reply:
@phoenixAja this is one of those elements I could not figure out how to inject data-test-id. From what I can gather, we might need to modify:

type DateMenuOption = {
  name: string; // Must be UNIQUE because we assume can be used as key/id
  // For both `numDays...` it's relative to now() when option is chosen.
  // Assume start is guaranteed, but end cap is not.
  numDaysEndOffset?: number; // How far back end of date interval is
  numDaysStartOffset?: number; // How far back start of date interval is
};

But I am not sure and I feel like I will break something if I fiddle around this